components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -182,6 +184,31 @@ open class GleanInternalAPI {
 
         // Set the OS type
         Baseline.os.set("Android")
+
+        // Set the OS version
+        // https://developer.android.com/reference/android/os/Build.VERSION
+        Baseline.osVersion.set(Build.VERSION.SDK_INT.toString())
+
+        // Set the device string
+        // https://developer.android.com/reference/android/os/Build
+        // We limit the device descriptor to 32 characters because it can get long. We give fewer
+        // characters to the manufacturer because we're less likely to have manufacturers with
+        // similar names than we are for a manufacturer to have two devices with the similar names
+        // (e.g. Galaxy S6 vs. Galaxy Note 6).
+        @Suppress("MagicNumber")
+        Baseline.device.set(
+            StringUtils.safeSubstring(Build.MANUFACTURER, 0, 12) +
+            '-' +
+            StringUtils.safeSubstring(Build.MODEL, 0, 19)
+        )
+
+        // Set the CPU architecture
+        val arch = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
```

Comment:
FYI, our minimum supported SDK version is 21, which _is_ LOLLIPOP, so this should always be true.

Reply:
Good catch!